### ISSUE TITLE
Support for Gradle 5

### ISSUE THREAD

**Issue by rpalcolea** [bmuschko/gradle-docker-plugin#665]
### Expected Behavior
The plugin should be usable in Gradle 5

### Current Behavior
When running tests of the plugin, I get the following output:

```
Unsupported type (org.ajoberstar.grgit.Person) found for field 'author' while constructing immutable class org.ajoberstar.grgit.Commit.
Immutable classes only support properties with effectively immutable types including:
- Strings, primitive types, wrapper types, Class, BigInteger and BigDecimal, enums
- classes annotated with @KnownImmutable and known immutables (java.awt.Color, java.net.URI)
- Cloneable classes, collections, maps and arrays, and other classes with special handling
  (java.util.Date and various java.time.* classes and interfaces)
Other restrictions apply, please see the groovydoc for ImmutableOptions for further details
```

### Context
Gradle 5 introduced groovy 2.5 which contains breaking changes on the way `@Immutable` is handled.

gradle-git is not maintained anymore, however grgit does and version https://github.com/ajoberstar/grgit/releases/tag/3.0.0-beta.1 contains fixes for their POGOs

### Steps to Reproduce (for bugs)
1) clone the repo
2) ./gradlew wrapper --gradle-version=5.0-20181002235907+0000
3) ./gradlew test

### Your Environment
This is environment agnostic

**Comment by bmuschko** [comment]
In the past I tried to upgrade the Gradle Git plugin and the relevant ecosystem. However, there have been radical changes to the way they are being used that I did not find well documented. I had shelved it but will have to pick it up again as soon as Gradle 5 comes out. Would you be interested in helping out with a pull request?

**Comment by rpalcolea** [comment]
Hi @bmuschko ,

Yeah, we faced the issue with grgit and nebula-release-plugin and ended up porting some code and upgrading grgit.

I'll add this to my plans and try to upgrade the plugin followed by a pull request

**Comment by rpalcolea** [comment]
Just heads up: 

https://issues.apache.org/jira/browse/GROOVY-8806?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=16636494
https://github.com/apache/groovy/pull/803/files

I opened an issue on Groovy JIRA and there is a fix for it. I already verified it. So as long as this gets released and Gradle moves into 2.5.3 (potential version for the fix), this plugin should be good. Same as Bintray  or others

**Comment by bmuschko** [comment]
Thanks, you may want to inform the Gradle project as well by opening an issue so they are aware of it.

**Comment by cdancy** [comment]
@rpalcolea yeah we'll have to do that eventually as that plugin has been broken up into multiple plugins and I believe is now sunset with the maintainer only taking in bugfixes/etc. We use his other plugins here internally with success we would just have to re-work some things in our scripts. Not bad when all is said and done but not urgent as the code works as-is for the time being.
